### DISCUSSION TITLE
Documentation Overhaul

### DISCUSSION THREAD

**Discussion by jrast** [FreeOpcUa/opcua-asyncio#1004] [Ideas]
Currently the documentation for the opcua-asyncio package still referes to [documentation of the python-opcua package](https://python-opcua.readthedocs.io/en/latest/) (see [here](https://github.com/FreeOpcUa/opcua-asyncio#documentation)). 

While most of the documentation might still apply to this package, it's slightly confusing as the "old" package was sync and now many parts are async.  Also for newcommers the documentation lacks examples, or at least links to examples. 

What do I propose:
- Create a new documentation for this package
- Include at least a complete (autogenerated) API documentation
- Include some examples to get started. The [examples in the repo](https://github.com/FreeOpcUa/opcua-asyncio/tree/master/examples) can be reused here and if @AndreasHeine allows we might also reuse some of his content in his [opcua-tutorial repo](https://github.com/AndreasHeine/opcua-tutorial)

Where can I help:
- I'm willing to put in some effort to get a initial version of the documentation

What do you think?
- Sound good?
- Any preffered Sphinx theme? I would suggest the [pydata theme](https://pydata-sphinx-theme.readthedocs.io/en/latest/examples/gallery.html)

**Comment by oroulet** [comment]
help is welcome yes. There is a docs folder in opcua-asyncio, still containing the python-opcua documenation. First thing to do is read the few files there and update them. Then one should find out how to generate and publish the doc. I did it last time but I completely forgot. 
you are welcome to change themes if you find something better

**Reply by jrast** [reply]
I will look through what's already there and reuse when possible. It's also a long time ago since I last set up a documentation deployment to ReadTheDocs, but I'm sure we will manage it. I'll start in a Docu branch in my fork and keep  you updated.

**Comment by AndreasHeine** [comment]
@jrast feel free to use it! the code has no licence on it 😉

**Reply by jrast** [reply]
Never assume that no licence means free to use, but thanks for the permission!

**Comment by jrast** [comment]
I started this PR: https://github.com/FreeOpcUa/opcua-asyncio/pull/1007
Currently there is not much to see, but creating the PR now allows everyone to see the state / progress (hopefully). I will also try to setup a RTD project to get a preview of the documentation.

**Reply by jrast** [reply]
Update: A preview is available here: https://opcua-asyncio-documentation.readthedocs.io/en/latest/index.html

Note: I choose the name of the RTD project to be "opcua-asyncio-documentation", this should make it easy to create a "opcua-asyncio" project once the documentation is integrated in this repo.
